packages/open-api-gateway/src/project/open-api-gateway-ts-project.ts

```diff
@@ -0,0 +1,243 @@
+    const codegenDirRelativeToParent = this.hasParent
+      ? path.join(options.outdir!, codegenDir)
+      : codegenDir;
+
+    // We generate the typescript client since this project will take a dependency on it
+    this.generatedTypescriptClient = new GeneratedTypescriptClientProject({
+      parent: this.hasParent ? options.parent : this,
+      defaultReleaseBranch: options.defaultReleaseBranch,
+      name: `${this.package.packageName}-typescript`,
+      outdir: path.join(codegenDirRelativeToParent, ClientLanguage.TYPESCRIPT),
+      // Use the parsed spec such that refs are resolved to support multi-file specs
+      specPath: spec.parsedSpecPath,
+      packageManager: options.packageManager,
+    });
+
+    // Synth early so that the generated code is available prior to this project's install phase
+    this.generatedTypescriptClient.synth();
+
+    const typescriptCodeGenDir = path.relative(
+      this.outdir,
+      this.generatedTypescriptClient.outdir
+    );
+
+    if (this.hasParent) {
+      // When we have a parent project, we can add a dependency on the generated typescript client since it's part of
+      // the monorepo. Since this project will be synthesized before the parent monorepo, the generated typescript
+      // client won't be visible for the first install in this project's post synthesize step, so we use a local
+      // workspace as a workaround.
+      if (this.package.packageManager === NodePackageManager.PNPM) {
+        new YamlFile(this, "pnpm-workspace.yaml", {
+          readonly: true,
+          obj: {
+            packages: [typescriptCodeGenDir],
+          },
+        });
+      } else {
+        this.package.addField("workspaces", {
+          packages: [typescriptCodeGenDir],
+        });
+      }
+      // Add the dependency
+      this.addDeps(this.generatedTypescriptClient.package.packageName);
+      // Since the generated typescript client project is parented by this project's parent rather than this project,
+      // projen will clean up the generated client when synthesizing this project unless we add an explicit exclude.
+      this.addExcludeFromCleanup(`${codegenDir}/**/*`);
+    } else {
+      // Add a file dependency on the generated typescript client
+      this.addDeps(
+        `${this.generatedTypescriptClient.package.packageName}@file:./${typescriptCodeGenDir}`
+      );
+
+      // Linting runs post synthesize before the typescript client has been built
+      this.eslint?.addRules({
+        "import/no-unresolved": [
+          2,
+          { ignore: [this.generatedTypescriptClient.package.packageName] },
+        ],
+      });
+    }
+
+    // Additional languages to generate other than typescript which is mandatory
+    const extraLanguages = new Set(options.clientLanguages);
+    extraLanguages.delete(ClientLanguage.TYPESCRIPT);
+    // TODO: generate clients in other given languages
+
+    // Generate the sample source and test code
```

Comment:
What if the user doesn't want to generate sample code?

Reply:
So I treat the "sampleCode" as `sample-api.ts` and `sayHello.handler.ts` which you can turn off with `sampleCode: false`, but `api.ts` and `api.test.ts` are the construct wrappers which provide the type safe interfaces, and I think it's better to always generate those :) What do you think?